sdk/src/test/java/com/horizen/account/transaction/EthereumTransactionRlpStreamCodecTest.java

```diff
@@ -164,9 +150,225 @@ public void rlpEthTxWithTrimmedBytes() {
+            byte[] b = BytesUtils.fromHexString(strings[1]);
+            Reader reader = new VLQByteBufferReader(ByteBuffer.wrap(b));
+            try {
+                EthereumTransaction ethTx = EthereumTransactionDecoder.decode(reader);
+                //ethTx.semanticValidity();
+                if (reader.remaining() != 0)
+                    throw new IllegalArgumentException("Stream is not consumed, we have " + reader.remaining() + "bytes left to read");
+                System.out.println(strings[0] + "--->" + ethTx.toString());
+                fail("Should not succeed");
+            } catch (Exception e) {
+                System.out.println(strings[0] + "--->" + e.getMessage());
+            }
+        }
+    }
+
+
+    @Test
+    public void testGethDecodeRlpSemanticValidationStream() {
+
+        for (String[] strings : gethTestVectorsSemanticValdationFailure) {
+            byte[] b = BytesUtils.fromHexString(strings[1]);
+            Reader reader = new VLQByteBufferReader(ByteBuffer.wrap(b));
+            try {
+                EthereumTransaction ethTx = EthereumTransactionDecoder.decode(reader);
+                ethTx.semanticValidity();
+                System.out.println(strings[0] + "--->" + ethTx);
+                fail("Should not succeed");
+
+            } catch (Exception e) {
+                System.out.println(strings[0] + "--->" + e.getMessage());
+            }
+        }
+    }
+
+    @Test
+    public void checkUnsignedTx() {
+        // https://github.com/ethereumbook/ethereumbook/blob/develop/code/web3js/raw_tx/raw_tx_demo.js
+        byte[] b = BytesUtils.fromHexString("e6808609184e72a0008303000094b0920c523d582040f2bcb1bd7fb1c7c1ecebdb3480801c8080");
+        Reader reader = new VLQByteBufferReader(ByteBuffer.wrap(b));
+        EthereumTransaction ethTx = EthereumTransactionDecoder.decode(reader);
+        assertFalse(ethTx.isSigned());
+
+        VLQByteBufferWriter writer = new VLQByteBufferWriter(new ByteArrayBuilder());
+        ethTx.encode(true, writer);
+        byte[] result = writer.toBytes();
+        assertEquals(BytesUtils.toHexString(b), BytesUtils.toHexString(result));
+    }
+
+    // useful when developing tests
+    @Test
+    @Ignore
+    public void checkSingleStream() throws TransactionSemanticValidityException {
+        byte[] b = BytesUtils.fromHexString("f86e09850a02ffee00825208943535353535353535353535353535353535353535880de0b6b3a7640000808200bda028ef61340bd939bc2195fe537567866003e1a15d3c71ff63e1590620aa636276a067cbe9d8997f761aecb703304b3800ccf555c9f3dc64214b297fb1966a3b6d83");
+        //byte[] b = BytesUtils.fromHexString(gethTestVectorsNotFailingWithW3j[2][1]);
+        Reader reader = new VLQByteBufferReader(ByteBuffer.wrap(b));
+        EthereumTransaction ethTx = EthereumTransactionDecoder.decode(reader);
+        ethTx.semanticValidity();
+        System.out.println(ethTx);
+    }
+
     @Test
     @Ignore
-    public void checkPerfStream() {
+    public void checkPerf() {
```

Comment:
I'm not sure if comparison has sense now,
the difference between 2 approaches is that the first one also get bytes from the write stream.
Maybe some changes to the test notes/prints needed. Or even can be removed fully 
What do you think?

Reply:
Ok, I removed this test too